h2/src/main/org/h2/result/LocalResult.java

```diff
@@ -318,15 +311,15 @@ public void addRow(Value[] values) {
             } else {
                 rowCount = external.addRow(values);
             }
-            return;
-        }
-        rows.add(values);
-        rowCount++;
-        if (rows.size() > maxMemoryRows) {
-            if (external == null) {
-                createExternalResult();
+        } else {
+            rows.add(values);
+            rowCount++;
+            if (rows.size() > maxMemoryRows) {
+                if (external == null) {
+                    createExternalResult();
+                }
+                addRowsToDisk();
```

Comment:
should maybe these 2 line inside addRowsToDisk()

    if (external == null) {
        createExternalResult();
   }

would make the logic cleaner and easier to read